benchmark/parser/main.go

```diff
@@ -0,0 +1,109 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+)
+
+type BenchmarkEvent struct {
+	Benchmark string `json:"benchmark"`
+	Event     string `json:"event"`
+	TestName  string `json:"test_name"`
+	UUID      string `json:"uuid"`
+	Timestamp string `json:"time"`
+}
+
+type ProfileEvent struct {
+	EventName string
+	StartTime time.Time
+	StopTime  time.Time
+}
+
+type BenchmarkProfile struct {
+	UUID     string
+	TestName string
+	Events   map[string]ProfileEvent
+}
+
+func main() {
+	logMap := make(map[string]BenchmarkProfile)
+	logFileName := os.Args[1]
+	logFile, err := os.Open(logFileName)
+	if err != nil {
+		panic("Cannot open " + logFileName)
+	}
+	fileScanner := bufio.NewScanner(logFile)
+
+	fileScanner.Split(bufio.ScanLines)
+
+	for fileScanner.Scan() {
+		line := fileScanner.Text()
+		if strings.Contains(line, "benchmark") {
+			parseLogLineToMap(fileScanner.Text(), logMap)
+		}
+	}
```

Comment:
After `for fileScanner.Scan()`, should we check the scanner error in case the scanning has errors and stops? Maybe something like this:

```golang
for fileScanner.Scan() { ... }

if err := fileScanner.Err(); err != nil {
  panic("scan error: " + err.Error())
}
```

Reply:
Done